test/unit/argument_parser/detail/version_check_test.hpp

```diff
@@ -46,24 +47,61 @@ struct version_check : public ::testing::Test
     char const * const OPTION_OFF = "0";
     char const * const OPTION_ON = "1";
 
-    std::chrono::duration<long int>::rep const TIME_NOW =
-        std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+    std::string const app_name = std::string{"test_version_check"};
 
-    std::string const APP_NAME = "test_version_check_" + app_name_component + "_" + std::to_string(TIME_NOW); // avoid name conflicts.
+    // This tmp_filename will create the file "version_checker.tmpfile" in a unique folder.
+    seqan3::test::tmp_filename tmp_file{"version_checker.tmpfile"};
 
-    std::filesystem::path const PATH = detail::version_checker::get_path();
+    void randomise_home_folder()
+    {
+        using namespace std::string_literals;
+        auto tmp_directory = tmp_file.get_path().parent_path();
 
-    std::filesystem::path const APP_VERSION_FILENAME = PATH / (APP_NAME + ".version");
+        auto result = setenv(detail::version_checker::home_env_name, tmp_directory.c_str(), 1);
+        if (result != 0)
+            throw std::runtime_error{"Couldn't set environment variable 'home_env_name' (="s +
+                                     detail::version_checker::home_env_name + ")"s};
 
-#if defined(NDEBUG)
-    std::filesystem::path const APP_TIMESTAMP_FILENAME = PATH / (APP_NAME + "_usr.timestamp");
-#else
-    std::filesystem::path const APP_TIMESTAMP_FILENAME = PATH / (APP_NAME + "_dev.timestamp");
-#endif // defined(NDEBUG)
+        auto is_prefix_path = [](std::string base_path, std::string path)
+        {
+            auto && it_pair = std::mismatch(base_path.begin(), base_path.end(), path.begin(), path.end());
+            return it_pair.first + 1 == base_path.end();
+        };
+
+        if (is_prefix_path(tmp_directory, app_tmp_path()))
+            throw std::runtime_error{"Setting the environment variable 'home_env_name' didn't have the correct effect"
+                                     " ("s + std::string{tmp_directory} + " is not a prefix of "s +
+                                     std::string{app_tmp_path()} + ")"s};
+    }
+
+    void SetUp() override
+    {
+        ::testing::Test::SetUp();
+
+        // set HOME environment to a random home folder before starting each test case
+        randomise_home_folder();
+    }
+
+    std::filesystem::path app_tmp_path()
+    {
+        return detail::version_checker::get_path();
+    }
+
+    std::filesystem::path app_version_filename()
+    {
+        return app_tmp_path() / (app_name + ".version");
```

Comment:
The division operator works here?

Reply:
Yes, this means concatenating two paths.

https://en.cppreference.com/w/cpp/filesystem/path/operator_slash

Reply:
The elegance of C++ 😄